### DISCUSSION TITLE
How do already installed programs and winget manifests correlate?

### DISCUSSION THREAD

**Discussion by gerardog** [microsoft/winget-cli#1949] [General] [answered]
There is a winget package (gerardog.gsudo) that installs `gsudoSetup.msi` from => https://github.com/gerardog/gsudo/releases/download/v1.0.2/gsudoSetup.msi

Is it possible for someone who installed the .msi directly (not using winget) to uninstall it using `winget uninstall gerardog.gsudo` ?

How do winget matches locally installed programs to existing winget manifests?

Thanks

**Comment by quhxl** [comment, marked as answer]
Yes, because winget doesn't care where the program has come from. It uses the values from Apps & features entry to identify the program and assign it to a package from the repo. Once this mapping is done, the program can be upgraded, uninstalled, etc. via the PackageIdentifier (gerardog.gsudo).

**Reply by gerardog** [reply]
Yes, but exactly how?

It's not working for gerardog.gsudo.
If I install the same .msi from outside winget, I can't upgrade or uninstall it.

So, how exactly do I link the `.msi` file to the package? 

Thx
